app/models/favorite_tag.rb

```diff
@@ -4,6 +4,11 @@ class FavoriteTag < ActiveRecord::Base
 
   validates :user_id, presence: true
   validates :tag_id, presence: true
+
+  validates_uniqueness_of :tag_id,
```

Comment:
Prefer the new style validations `validates :column, uniqueness: value` over `validates_uniqueness_of`.